openbook_posts/models.py

```diff
@@ -240,21 +240,31 @@ class PostVideo(models.Model):
 
 class PostComment(models.Model):
     post = models.ForeignKey(Post, on_delete=models.CASCADE, related_name='comments')
+    parent_comment = models.ForeignKey('self', on_delete=models.CASCADE, related_name='replies', null=True, blank=True)
     created = models.DateTimeField(editable=False, db_index=True)
     commenter = models.ForeignKey(User, on_delete=models.CASCADE, related_name='posts_comments')
     text = models.CharField(_('text'), max_length=settings.POST_COMMENT_MAX_LENGTH, blank=False, null=False)
     is_edited = models.BooleanField(default=False, null=False, blank=False)
 
     @classmethod
-    def create_comment(cls, text, commenter, post):
-        return PostComment.objects.create(text=text, commenter=commenter, post=post)
+    def create_comment(cls, text, commenter, post, parent_comment=None):
+        return PostComment.objects.create(text=text, commenter=commenter, post=post, parent_comment=parent_comment)
 
     @classmethod
     def count_comments_for_post_with_id(cls, post_id):
-        count_query = Q(post_id=post_id)
+        count_query = Q(post_id=post_id, parent_comment__isnull=True)
+
+        return cls.objects.filter(count_query).count()
+
+    @classmethod
+    def count_comment_replies_for_post_with_id_for_comment_with_id(cls, post_id, post_comment_id):
+        count_query = Q(post_id=post_id, parent_comment__id=post_comment_id)
 
         return cls.objects.filter(count_query).count()
 
+    def reply_to_comment(self, commenter, post, text):
```

Comment:
So our comment api has /:post_uuid/comments/:comment_id   ... we can ignore the post_uuid everywhere and then there would be no need to check if a comment belongs to post, since the post will be derived directly from it ... 
or ..
we use this post_uuid , and do the check.. since we were already doing this elsewhere I thought for consistency i'd keep it like that

Reply:
This is the `PostComment` model, on the `User` model, there is no relation between a `post_uuid` and a `comment_id`, here the relation is already given as a own attribute. No need to pass the post here, this could lead to having replies to comments on a post that is not the owner of the comment.

Reply:
Is like passing the user to a method on the `UserProfile` model. The UserProfile model cannot exist without the user, its redundant and can lead to wrong data.